examples/cpp/dds/AdvanceConfigurationExample/arg_configuration.h

```diff
@@ -0,0 +1,216 @@
+    { DOMAIN_ID, 0, "d", "domain",                Arg::Numeric,
+      "  -d <id> \t--domain=<id>  \tDDS domain ID (Default: 0)." },
+    { WAIT, 0, "w", "wait",                 Arg::Numeric,
+      "  -w <num> \t--wait=<num> \tNumber of matched subscribers required to publish"
+      " (Default: 0 => does not wait)." },
+    { SAMPLES, 0, "s", "samples",              Arg::Numeric,
+      "  -s <num> \t--samples=<num>  \tNumber of samples to send (Default: 0 => infinite samples)." },
+    { INTERVAL, 0, "i", "interval",            Arg::Numeric,
+      "  -i <num> \t--interval=<num>  \tTime between samples in milliseconds (Default: 100)." },
+    { ASYNC, 0, "a", "async",               Arg::None,
+      "  -a \t--async \tAsynchronous publish mode (synchronous by default)." },
+    { TRANSPORT, 0, "", "transport",        Arg::Transport,
+      "  \t--transport=<shm|udp|udpv6> \tUse only shared-memory, UDPv4, or UDPv6 transport."
+      "If not set, use Fast DDS default transports (depending on the scenario it will use the most efficient one:"
+      " data-sharing delivery mechanism > shared-memory > UDP)." },
+    { OWNERSHIP, 0, "o", "ownership",        Arg::None,
+      "  -o \t--ownership \tUse Topic with OWNERSHIP_EXCLUSIVE."},
+    { OWNERSHIP_STRENGTH, 0, "", "strength",        Arg::Numeric,
+      "  \t--strength=<num> \tSet this Publisher strength. Set Topic with OWNERSHIP_EXCLUSIVE. Default: 0"},
+
+    { UNKNOWN_OPT, 0, "", "",                Arg::None,      "\nSubscriber options:"},
+    { TOPIC, 0, "t", "topic",                  Arg::String,
+      "  -t <topic_name> \t--topic=<topic_name>  \tTopic name (Default: HelloWorldTopic)." },
+    { DOMAIN_ID, 0, "d", "domain",                Arg::Numeric,
+      "  -d <id> \t--domain=<id>  \tDDS domain ID (Default: 0)." },
+    { SAMPLES, 0, "s", "samples",              Arg::Numeric,
+      "  -s <num> \t--samples=<num>  \tNumber of samples to wait for (Default: 0 => infinite samples)." },
+    { TRANSPORT, 0, "", "transport",        Arg::Transport,
+      "  \t--transport=<shm|udp|udpv6> \tUse only shared-memory, UDPv4, or UDPv6 transport."
+      "If not set, use Fast DDS default transports (depending on the scenario it will use the most efficient one:"
+      " data-sharing delivery mechanism > shared-memory > UDP)." },
+    { OWNERSHIP, 0, "o", "ownership",        Arg::None,
+      "  -o \t--ownership \tUse Topic with OWNERSHIP_EXCLUSIVE."},
+
+    { UNKNOWN_OPT, 0, "", "",                Arg::None,      "\nQoS options:"},
+    { RELIABLE, 0, "r", "reliable",         Arg::None,
+      "  -r \t--reliable \tSet reliability to reliable (best-effort by default)." },
+    { TRANSIENT_LOCAL, 0, "", "transient",        Arg::None,
+      "  \t--transient \tSet durability to transient local (volatile by default, ineffective when not reliable)." },
+    { PARTITIONS, 0, "p", "partitions",        Arg::String,
+      "  -p <str> \t--partitions=<str> \tPartitions to match separated by ';'. "
```

Comment:
NIT: When having more than one partition, this argument has to be enclosed by simple or double quotes to be correctly parsed. Maybe we should warn about this.